charmcraft/commands/store/store.py

```diff
@@ -211,6 +211,13 @@ def create_library_revision(self, charm_name, lib_id, api, patch, content, conte
         result = _build_library(response)
         return result
 
+    def get_library(self, charm_name, lib_id, api):
+        """Get the library tip by id for a given api version."""
+        endpoint = '/v1/charm/libraries/{}/{}?api={}'.format(charm_name, lib_id, api)
```

Comment:
question, is there any charm_name or lib_id that would ever need to be url encoded? I suspect not

Reply:
Right, both names are validated to be a small subset of ASCII.